From: Paul Brook <paul@nowt.org>
Subject: BF: specify explicit time offset not a time zone name to avoid needing tzdata during testing

Origin: Debian
Bug-Debian: https://bugs.debian.org/855920
Forwarded: https://github.com/fail2ban/fail2ban/pull/1754
Applied-Upstream: yes
Last-Update: 2017-04-17


--- a/fail2ban/tests/utils.py
+++ b/fail2ban/tests/utils.py
@@ -95,7 +95,10 @@ def setUpMyTime():
 	# Set the time to a fixed, known value
 	# Sun Aug 14 12:00:00 CEST 2005
 	# yoh: we need to adjust TZ to match the one used by Cyril so all the timestamps match
-	os.environ['TZ'] = 'Europe/Zurich'
+	# This offset corresponds to Europe/Zurich timezone.  Specifying it
+	# explicitly allows to avoid requiring tzdata package to be installed during
+	# testing.   See https://bugs.debian.org/855920 for more information
+	os.environ['TZ'] = 'CET-01CEST-02,M3.5.0,M10.5.0'
 	time.tzset()
 	MyTime.setTime(1124013600)
 
