Index: fail2ban/config/paths-debian.conf
===================================================================
--- fail2ban.orig/config/paths-debian.conf
+++ fail2ban/config/paths-debian.conf
@@ -26,3 +26,6 @@ exim_main_log = /var/log/exim4/mainlog
 # was in debian squeezy but not in wheezy
 # /etc/proftpd/proftpd.conf (SystemLog)
 proftpd_log = /var/log/proftpd/proftpd.log
+
+# Debian specific path
+roundcube_errors_log = /var/log/roundcube/errors.log
