Index: fail2ban/config/filter.d/sshd.conf
===================================================================
--- fail2ban.orig/config/filter.d/sshd.conf
+++ fail2ban/config/filter.d/sshd.conf
@@ -93,7 +93,7 @@ mdre-aggressive = %(mdre-ddos)s
 mdre-aggressive-other = %(mdre-ddos-other)s
 
 # Parameter "publickey": nofail (default), invalid, any, ignore
-publickey = nofail
+publickey = any
 # consider failed publickey for invalid users only:
 cmnfailre-failed-pub-invalid = ^Failed publickey for invalid user <F-USER>(?P<cond_user>\S+)|(?:(?! from ).)*?</F-USER> from <HOST>%(__on_port_opt)s(?: ssh\d*)?(?(cond_user): |(?:(?:(?! from ).)*)$)
 # consider failed publickey for valid users too (don't need RE, see cmnfailed):
